python/ray/serve/_private/router.py

```diff
@@ -359,15 +410,9 @@ async def assign_request(
         finally:
             # If the request is disconnected before assignment, this coroutine
             # gets cancelled by the caller and an asyncio.CancelledError is
-            # raised. The finally block ensures that num_queued_queries
+            # raised. The finally block ensures that num_queued_requests
             # is correctly decremented in this case.
-            self.num_queued_queries -= 1
-            self.num_queued_queries_gauge.set(self.num_queued_queries)
+            self._metrics_manager.dec_num_queued_requests()
```

Comment:
I'd suggest to add a context manager method to the metrics manager instead, so the body of this method would look like:
```
with self._metrics_manager.wrap_request_assignment():
    ... existing logic ...
```
and the `try: finally:` logic would live inside it along with the gauges